python/mechanism.cpp

```diff
@@ -197,7 +198,7 @@ void register_mechanisms(pybind11::module& m) {
     m.def("default_catalogue", [](){return arb::global_default_catalogue();});
     m.def("allen_catalogue", [](){return arb::global_allen_catalogue();});
     m.def("bbp_catalogue", [](){return arb::global_bbp_catalogue();});
-    m.def("load_catalogue", [](const std::string& fn){return arb::load_catalogue(fn);});
+    m.def("load_catalogue", [](pybind11::object fn) { return arb::load_catalogue(util::to_string(fn)); });
```

Comment:
```suggestion
    m.def("load_catalogue", [](pybind11::object fn) { return arb::load_catalogue(util::to_path(fn)); });
```
?

Reply:
When is `std::fs`` due?

Reply:
C++17, it's supported by all compilers, **but** MacOS pre-10.15 doesn't have the proper syscalls, so there **no** compiler has it. @espenhgn had issues with that and we reverted.

Reply:
Hi. I've since gotten another machine (w. macOS 11.6). I've had no issues compiling Arbor-v0.6 on that. If you now keep 10.15 support alive only for me, feel free to revert back.

Reply:
We've prevented `std::filesystem` from entering the Arbor code base for this reason, and it is good to know that we now can :)